sandbox.go

```diff
@@ -263,6 +266,39 @@ func (sb *sandbox) updateGateway(ep *endpoint) error {
 	return nil
 }
 
+func (sb *sandbox) SetKey(basePath string) error {
+	var err error
+	if basePath == "" {
+		sb.Unlock()
+		return fmt.Errorf("invalid sandbox key")
+	}
+
+	sb.Lock()
+	if sb.osSbox != nil {
+		sb.Unlock()
+		return fmt.Errorf("failed to set sandbox key : already assigned")
+	}
+	endpoints := sb.endpoints
```

Comment:
Getting a reference to the sandbox endpoint list under sandbox lock would not give you any peace of mind when you later will iterate trough the referenced list outside of lock.

Please remove this and use `getConnectedEndpoints()` later on.